### ISSUE TITLE
Resist fingerprinting interfere with the addon

### ISSUE THREAD

**Issue by 0brien** [mozilla/multi-account-containers#2276]
TL;DR: does using Multi-Account Containers add-on in order to benefit from MozillaVPN integration mess up anything important in RFP (or the Arkenfox UserJS)?

Hi all,

I have a couple of questions about the RFP (which I use as part of the Arkenfox UserJS) and the Multi-Account Containers (MAC) add-on. I don’t know how familiar people are with the Arkenfox UserJS but in case anyone isn’t familiar, it’s worth mentioning it has RFP on and also uses dFPI. (Not sure if there are any other key considerations to mention – if there are, please do say so!)

In short, I use the MAC add-on rather than using the containers built into the FF settings so that each container can use a different VPN with MozillaVPN. However, I noticed that when RFP is enabled I can’t add MAC. It tells me ‘You need an updated version of Firefox for this extension.’

If I disable RFP, I can add MAC, and if I later re-enable RFP I can keep MAC and it seems to work with no actual issues as far as I can tell. My guess is that the reason for the error message is that RFP spoofs the user agent, giving an old version of FF. So as a result the FF add-ons store thinks MAC is not compatible with my install of FF, but in reality it is.

My questions are:

i) Is my understanding of why I am getting this error message correct or is there some actual incompatibility with RFP?

ii) is there a risk of anything going wrong, either with MAC or with RFP or the Arkenfox UserJS more broadly, from a privacy or security POV as a result of my using them together? I normally wouldn’t think twice but the fact that the extension doesn’t show up when I have RFP enabled makes me wonder if it is something more than just the spoofed user agent that is an issue. (One example that comes to mind - I wonder if MAC might interfere with dFPI that Arkenfox enables.)

Thanks in advance!

**Comment by zefilipe** [comment]
Hi,

Im using very well BitWarden, with Multi-account containers, but i use a professional VPN that isnt MozillaVPN !

**Comment by dannycolin** [comment]
> My guess is that the reason for the error message is that RFP spoofs the user agent, giving an old version of FF. So as a result the FF add-ons store thinks MAC is not compatible with my install of FF, but in reality it is.

That's most likely what's happening here. IIRC, it spoofs it with the latest ESR version but I'd have to test it to be sure. If it doesn't it still should work. However, I remember that there was a bug in AMO itself that in some situation prevents users running ESR to install the addon and I guess you might simply have triggered it by using RFP.

**Comment by Thorin-Oakenpants** [comment]
FYI: since FF102, RFP no longer spoofs versioning on **_desktop_** - it still does on Android (as ESR) - https://hg.mozilla.org/mozilla-central/rev/89e459d1b6cc. For Tor Browser desktop they are based on ESR so it has no effect, and for Firefox users with RFP it means the versioning is correct. Note: trying to hide your version is [pointless](https://arkenfox.github.io/TZP/tests/versions.html)

**Comment by dannycolin** [comment]
@Thorin-Oakenpants Thanks for the pointer. 

So when it was reported, RFP was spoofing the browser version to 91.0 and Multi-Account Containers was asking for 91.1.0 as the minimum version needed.

Since RFP doesnt spoof the version and that anyway the latest ESR is 102, this issue can be marked as resolved.
